util/shell/lex.go

```diff
@@ -501,6 +501,35 @@ func (sw *shellWord) processDollarCurlyBracket() (string, error) {
 			return "", errors.Errorf("%s: %s", name, message)
 		}
 		return newValue, nil
+	case '#':
```

Comment:
If I'm not mistaken, this case appears to be identical to the above `case '?'`, except for the returned error message.

If so, can you combine them rather than duplicate this code?

Reply:
There's one subtle difference [here](https://github.com/earthly/earthly/pull/3529/files#diff-bc232306ec1e9b099303c5304a7ffd0d3783f0d39aa9430d1c424a1330cd7614R524).

But I can probably create a shared helper function if it's worthwhile.

Reply:
I totally read that as `fmt.Errorf` :sweat_smile: 

I don't think a helper function is needed, couldn't you just use the value of `ch`?

e.g.

`fmt.Sprintf("${%s#%s}", name, word)` -> `fmt.Sprintf("${%s%s%s}", name, ch, word)`

Reply:
Oh, good point. Will do.

Reply:
Actually, the previous code was incorrect. I misunderstood. Updated with correct behavior.